lib/dish_and_menu.rb

```diff
@@ -0,0 +1,169 @@
+            puts "1. Starters"
+            puts "2. Mains"
+            puts "3. Desserts"
+            puts "4. Checkout"
+            input = gets.chomp.to_i
+            if input == 4
+                checkout(basket)
+                break 
+            else
+                display_menu(input - 1)
+                basket << select_dish(input - 1)
+            end
+        end
+    end
+
+    def checkout(basket)
+        total = 0
+        basket.each do |dish|
+            total += dish.price.to_f
+        end
+        print_receipt(basket, total)
+        if check_order?
+            puts "Thank you for your order! You will soon receive a SMS confirmation".green
+            send_text_confirmation
+            return total
+        else
+            puts "Your order has been cancelled.".red
+            return nil
+        end
+    end
+
+    def shopping_menu
+        while true
+            system "clear"
+            puts "1. View menu"
+            puts "2. Quit"
+            input = gets.chomp
+            case input
+            when "1"
+                menu_options
+            when "2"
+                break
+            end
+        end
+        system "clear"
+        puts "See you later!".green
+        nil
+    end
+
+    def send_text_confirmation
+        account_sid = ENV["TWILIO_SID"]
+        auth_token = ENV["AUTH_TOKEN"]
+
+        @client = Twilio::REST::Client.new account_sid, auth_token
+        message = @client.messages.create(
+            body: "Thank you for making your order, it's on its way!",
+            to: ENV["PHONE_NO"],
+            from: "+14328887495")
+
+        if message.sid
+            true
+        else
+            false
+        end
+    end
+
+    def load_dishes
+        file = File.open("dishes.csv", "a+")
+        file.readlines.each do |line|
+            name, price, course = line.chomp.split(",")
+            @dishes[course.to_i] << Dish.new(name, price)
+        end
+        file.close
+    end
+
+    def save_dishes
+        file = File.open("dishes.csv", "w+")
+        @dishes.each_with_index do |course, index|
+            course.each do |dish|
+                dish_data = [dish.name, dish.price, index].join(",")
+                file.puts dish_data
+            end
+        end
+        file.close
+    end
+
+    private
+    def print_receipt(basket, total)
+        printed_dishes = []
+        system "clear"
+        puts "Order:"
+        basket.each do |dish|
+            if !printed_dishes.include?(dish)
+                amount = basket.count(dish)
+                puts "#{dish.name} X#{amount} -- £#{dish.price * amount}"
+                printed_dishes << dish
+            end
+        end
+        puts "Total: £#{total}"
+    end
+
+    def check_order?
+        puts "Confirm order Y/N:"
+        input = gets.chomp
+        case input.downcase
+        when "y"
+            true
+        when "n"
+            false
+        end
+    end
+
+
+
+end
```

Comment:
I really like the idea of keeping the menu in the csv file, that's a nice extra work you did here. 
however, why are all the classes kept in the single .rb file?